spec/libraries/course/assessment/programming_test_case_report_spec.rb

```diff
@@ -202,6 +202,124 @@
     end
   end
 
+  # Test dynamic hints, failure and error messages, messages hash,
+  # output meta
+  context 'when given a report with various kinds of output data' do
+    let(:report_path) do
+      File.join(Rails.root, 'spec/fixtures/course/'\
+                'programming_messages_test_report.xml')
+    end
+
+    let(:report_xml) { File.read(report_path) }
+
+    let(:parsed_report) do
+      Course::Assessment::ProgrammingTestCaseReport.new(report_xml)
+    end
+    let(:public_test_cases) { parsed_report.test_suites.first.test_cases }
+    let(:private_test_cases) { parsed_report.test_suites.second.test_cases }
+
+    describe Course::Assessment::ProgrammingTestCaseReport::TestCase do
+      describe 'failed test case' do
```

Comment:
Use `context` for branches. e.g. `context 'when the test case failed'`,  Same for other places.